modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -652,14 +698,22 @@ void cv::gimpl::ie::RequestPool::execute(cv::gimpl::ie::RequestPool::Task&& t) {
 
     auto& request = m_requests[id];
 
+    using namespace std::placeholders;
+    using callback_t = std::function<void(IE::InferRequest, IE::StatusCode)>;
     request.SetCompletionCallback(
-            std::bind(&cv::gimpl::ie::RequestPool::callback, this, t, std::ref(request), id));
+            static_cast<callback_t>(
+                std::bind(&cv::gimpl::ie::RequestPool::callback, this,
+                          t, id, _1, _2)));
     t.run(request);
 }
 
 void cv::gimpl::ie::RequestPool::callback(cv::gimpl::ie::RequestPool::Task task,
-                                          InferenceEngine::InferRequest& request,
-                                          size_t id) {
+                                          size_t id,
+                                          IE::InferRequest request,
+                                          IE::StatusCode /* code */) {
+    // FIXME: Ignore IE::StatusCode so far,
```

Comment:
@dmatveev Does it make sense to put some log here in order to notify user about problem?